File: packages/android/src/main/java/com/kozmos/components/RouteOptionCard/RouteOptionCard.kt
package com.kozmos.components.routeoptioncard

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.defaultMinSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.Accessible
import androidx.compose.material.icons.filled.Schedule
import androidx.compose.material.icons.filled.Tune
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.semantics.contentDescription
import androidx.compose.ui.semantics.selected
import androidx.compose.ui.semantics.semantics
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.unit.dp
import com.kozmos.contracts.KozmosRouteOptionPresentation
import com.kozmos.contracts.KozmosRoutePreference
import com.kozmos.tokens.KozmosColors
import com.kozmos.tokens.KozmosDimensions

/**
 * A selectable route option.
 *
 * Mirrors the React `RouteOptionCard`. The card never changes its own selected
 * state — it reports selection and re-renders from the supplied presentation.
 */
@Composable
fun KozmosRouteOptionCard(
    option: KozmosRouteOptionPresentation,
    onSelect: (String) -> Unit,
    modifier: Modifier = Modifier,
    enabled: Boolean = true,
    icon: (@Composable () -> Unit)? = null
) {
    val isEnabled = enabled && option.available
    val accessibilityDescription = listOfNotNull(
        option.label,
        option.durationLabel,
        option.distanceLabel,
        option.warning
    ).joinToString(", ")

    Surface(
        onClick = { onSelect(option.id) },
        modifier = modifier
            .fillMaxWidth()
            .defaultMinSize(minHeight = 96.dp)
            .semantics {
                contentDescription = accessibilityDescription
                selected = option.selected
            },
        enabled = isEnabled,
        shape = RoundedCornerShape(KozmosDimensions.semanticsRadiusPanel),
        color = if (option.selected) {
            KozmosColors.primitivesColorsTheme500.copy(alpha = 0.05f)
        } else {
            KozmosColors.primitivesColorsBackground0
        },
        border = BorderStroke(
            width = if (option.selected) 2.dp else 1.dp,
            color = if (option.selected) {
                KozmosColors.primitivesColorsTheme500
            } else {
                KozmosColors.primitivesColorsForeground300
            }
        )
    ) {
        Column(
            modifier = Modifier.padding(KozmosDimensions.primitivesLayoutSpacing150),
            verticalArrangement = Arrangement.spacedBy(KozmosDimensions.primitivesLayoutSpacing150)
        ) {
            Row(
                verticalAlignment = Alignment.CenterVertically,
                horizontalArrangement = Arrangement.spacedBy(KozmosDimensions.primitivesLayoutSpacing100)
            ) {
                if (icon != null) {
                    icon()
                } else {
                    Icon(
                        imageVector = when (option.preference) {
                            KozmosRoutePreference.Quickest -> Icons.Default.Schedule
                            KozmosRoutePreference.StepFree -> Icons.AutoMirrored.Filled.Accessible
                            KozmosRoutePreference.Custom -> Icons.Default.Tune
                        },
                        contentDescription = null,
                        tint = KozmosColors.primitivesColorsTheme500
                    )
                }

                Text(
                    text = option.label,
                    style = MaterialTheme.typography.bodyMedium,
                    fontWeight = FontWeight.SemiBold,
                    color = KozmosColors.primitivesColorsForeground100,
                    maxLines = 1,
                    overflow = TextOverflow.Ellipsis
                )
            }

            Row(
                modifier = Modifier.fillMaxWidth(),
                verticalAlignment = Alignment.Bottom,
                horizontalArrangement = Arrangement.SpaceBetween
            ) {
                Text(
                    text = option.durationLabel,
                    style = MaterialTheme.typography.titleLarge,
                    fontWeight = FontWeight.SemiBold,
                    color = KozmosColors.primitivesColorsForeground100
                )

                Text(
                    text = option.distanceLabel,
                    style = MaterialTheme.typography.bodySmall,
                    color = KozmosColors.primitivesColorsForeground500
                )
            }

            option.warning?.let { warning ->
                Text(
                    text = warning,
                    style = MaterialTheme.typography.bodySmall,
                    color = KozmosColors.primitivesColorsEmotionalAlert600
                )
            }
        }
    }
}

File: packages/ios/Sources/Components/RouteOptionCard/RouteOptionCard.swift
import SwiftUI

/// A selectable route option.
///
/// Mirrors the React `RouteOptionCard`. The card never changes its own selected
/// state — it reports selection and re-renders from the supplied presentation.
public struct KozmosRouteOptionCard<Icon: View>: View {
    private let option: KozmosRouteOptionPresentation
    private let isDisabled: Bool
    private let onSelect: (String) -> Void
    private let icon: Icon?

    public init(
        option: KozmosRouteOptionPresentation,
        isDisabled: Bool = false,
        onSelect: @escaping (String) -> Void,
        @ViewBuilder icon: () -> Icon
    ) {
        self.option = option
        self.isDisabled = isDisabled
        self.onSelect = onSelect
        self.icon = icon()
    }

    private var disabled: Bool { !option.available || isDisabled }

    private var accessibilityDescription: String {
        [option.label, option.durationLabel, option.distanceLabel, option.warning]
            .compactMap { $0 }
            .joined(separator: ", ")
    }

    public var body: some View {
        Button {
            onSelect(option.id)
        } label: {
            VStack(alignment: .leading, spacing: 0) {
                HStack(spacing: KozmosDimensions.primitivesLayoutSpacing100) {
                    if let icon {
                        icon
                            .foregroundColor(KozmosColors.primitivesColorsTheme500)
                            .accessibilityHidden(true)
                    }

                    Text(option.label)
                        .font(.subheadline.weight(.semibold))
                        .lineLimit(1)
                        .truncationMode(.tail)
                }

                HStack(alignment: .bottom) {
                    Text(option.durationLabel)
                        .font(.title3.weight(.semibold))

                    Spacer(minLength: KozmosDimensions.primitivesLayoutSpacing150)

                    Text(option.distanceLabel)
                        .font(KozmosTypography.caption)
                        .foregroundColor(KozmosColors.primitivesColorsForeground500)
                }
                .padding(.top, KozmosDimensions.primitivesLayoutSpacing150)

                if let warning = option.warning {
                    Text(warning)
                        .font(KozmosTypography.caption)
                        .foregroundColor(KozmosColors.primitivesColorsEmotionalAlert600)
                        .fixedSize(horizontal: false, vertical: true)
                        .padding(.top, KozmosDimensions.primitivesLayoutSpacing100)
                }
            }
            .frame(maxWidth: .infinity, minHeight: 96, alignment: .leading)
            .foregroundColor(KozmosColors.primitivesColorsForeground100)
            .padding(KozmosDimensions.primitivesLayoutSpacing150)
            .background(
                option.selected
                    ? KozmosColors.primitivesColorsTheme500.opacity(0.05)
                    : KozmosColors.primitivesColorsBackground0
            )
            .clipShape(RoundedRectangle(cornerRadius: KozmosDimensions.semanticsRadiusPanel, style: .continuous))
            .overlay(
                RoundedRectangle(cornerRadius: KozmosDimensions.semanticsRadiusPanel, style: .continuous)
                    .stroke(
                        option.selected
                            ? KozmosColors.primitivesColorsTheme500
                            : KozmosColors.primitivesColorsForeground300,
                        lineWidth: option.selected ? 2 : 1
                    )
            )
        }
        .buttonStyle(.plain)
        .disabled(disabled)
        .opacity(disabled ? 0.5 : 1)
        .accessibilityLabel(accessibilityDescription)
        .accessibilityAddTraits(option.selected ? [.isButton, .isSelected] : .isButton)
    }
}

public extension KozmosRouteOptionCard where Icon == Image {
    /// Uses the SF Symbol that matches the option's routing preference.
    init(
        option: KozmosRouteOptionPresentation,
        isDisabled: Bool = false,
        onSelect: @escaping (String) -> Void
    ) {
        let systemImage: String
        switch option.preference {
        case .quickest: systemImage = "clock"
        case .stepFree: systemImage = "figure.roll"
        case .custom: systemImage = "slider.horizontal.3"
        }

        self.init(option: option, isDisabled: isDisabled, onSelect: onSelect) {
            Image(systemName: systemImage)
        }
    }
}
